### ISSUE TITLE
`transcript` is always empty

### ISSUE THREAD

**Issue by iamtheef** [JamesBrill/react-speech-recognition#127]
transcript doesn't seem to update at all

**Comment by JamesBrill** [comment]
Hi @iamtheef Could you provide more information? Do you have a code snippet you can share? And what browser are you running it on?

**Comment by iamtheef** [comment]
Sure!
Here's my button ` <button
        className="yn-button listen-button"
        name="submitButton"
        onMouseDown={() =>
          SpeechRecognition.startListening({ continuous: true })
        }
        onMouseUp={SpeechRecognition.stopListening}
      >`

And here's
`  useEffect(() => {
    console.log("listening", listening);
    console.log("is mic available", isMicrophoneAvailable);
    console.log("browser supports", browserSupportsSpeechRecognition);
    console.log("length", transcript.length);
    console.log("===");
  }, [listening]);`

**Comment by JamesBrill** [comment]
> My browser is Brave Version 1.31.91 Chromium: 95.0.4638.69 (Official Build) (x86_64) btw

I suspect this is the problem (your code looks fine). Although Chrome offers speech recognition, it is disabled in browsers like Brave that use Chromium. This is because Google owns the servers that provide that functionality - they can afford to provide it for free in their own browser, but other browser vendors would have to pay them to add that feature themselves. You can see discussions from unhappy Brave developers [here](https://github.com/brave/brave-browser/issues/3725) and [here](https://github.com/brave/brave-browser/issues/2802).

Out of interest, what value does [browserSupportsSpeechRecognition](https://github.com/JamesBrill/react-speech-recognition/blob/master/docs/API.md#browsersupportsspeechrecognition-bool) have on Brave? Ideally, it would be `false`. Given that you saw the red dot appear, I've got a feeling we're not detecting Brave's lack of speech recognition support correctly.

**Comment by iamtheef** [comment]
Thanks for the reply. This makes sense actually. So in other words `react-speech-recognition` only would be able to work on Google Chrome?
I suspected that detection doesn't work well too, `browserSupportsSpeechRecognition` is always true as presented in the console in the code above.

What would be another approach for speech recognition in a good amount of browsers?

**Comment by JamesBrill** [comment]
It is exactly this reason why I encourage users of `react-speech-recognition` to [integrate a polyfill](https://github.com/JamesBrill/react-speech-recognition#why-you-should-use-a-polyfill-with-this-library). The Web Speech API is supported in a handful of other browsers (e.g. recent versions of Safari), but the coverage still isn't great and each browser vendor has their own implementation - the polyfills bridge that gap and provide a consistent experience across browsers.

That detection can be improved - I shall make a new issue for failing more gracefully in Chromium browsers. Thanks for bringing that to my attention.

**Comment by iamtheef** [comment]
Gonna try implementing it with a polyfill then. Thanks a lot for your time!

**Comment by iamtheef** [comment]
@JamesBrill I managed to get it working eventually with polyfill. Another problem I noticed is that there are CORS problems when the polyfill gets initialised sometimes. It's really weird cause I noticed the problem when I first deployed the app with the pollyfill, then it went away for a few days and now it stopped working again. This only happens on the desktop app, it works fine on mobile.
Here's the error I am getting on the desktop app: 
`Access to fetch at 'https://api.speechly.com/login' from origin 'https://myapp.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by iamtheef** [comment]
And just now the error disappeared again after a few refreshes on the page, I think the login fails sometimes hence the error.

**Comment by JamesBrill** [comment]
@iamtheef I am surprised to see that error - has it disappeared consistently now? If not, it might be worth testing the [Speechly client](https://github.com/speechly/speechly/tree/main/libraries/browser-client) directly and see if you get the same error. If so, you could try raising an issue [here](https://github.com/speechly/speechly/tree/main/libraries/browser-client) as this is the library that initiates the Speechly login.

**Comment by nuri35** [comment]
I solved this problem by watching this video https://www.youtube.com/watch?v=sLXcuCrBL-M&t=363s
